### PR TITLE
Feature flag to make `readthedocs` theme default on MkDocs docs

### PR DESCRIPTION
**Pull request by humitos**
Historically, we were using `readthedocs` as default theme for MkDocs but in https://github.com/rtfd/readthedocs.org/pull/4556 we decided to change it to get the same behavior when building locally than in Read the Docs.

This commit adds a Feature flag to keep having the old behavior for some particular projects so we can add these project and do not break their documentation (change the theme without asking/reason).

### CODE REVIEW

File: readthedocs/doc_builder/backends/mkdocs.py [hunk 1]
```diff
@@ -50,6 +51,23 @@ def __init__(self, *args, **kwargs):
         self.root_path = self.version.project.checkout_path(self.version.slug)
         self.yaml_file = self.get_yaml_config()
 
+        # README: historically, the default theme was ``readthedocs`` but in
+        # https://github.com/rtfd/readthedocs.org/pull/4556 we change it to
+        # ``mkdocs`` to maintain the same behavior in Read the Docs than
+        # building locally. Although, we can't apply the this into the Corporate
```
Comment (stsewd) [inline readthedocs/doc_builder/backends/mkdocs.py]: `the this` -> `this`

**Comment by davidfischer** [comment]
* In MkDocs >= 1.0, this will never be a problem as theme is required. Therefore the default will never be used.
* In MkDocs >= 0.17 but < 1.0, the theme can be specified in [mkdocs.yml as a dict](https://www.mkdocs.org/about/release-notes/#theme-customization-1164) although it is optional. If the user specified it, I believe this PR won't do anything.
* In MkDocs < 0.17, the theme was always just a string although it was optional. This should handle that just fine.

File: readthedocs/projects/migrations/0028_feature-flag-mkdocs-theme.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('projects', '0027_remove_json_with_html_feature'),
+    ]
+
+    operations = [
+        migrations.RunPython(forward_add_feature, reverse_add_feature),
+    ]
```
[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/projects/migrations/0028_feature-flag-mkdocs-theme.py]: Well, the migration in this case only creates the Feature flag object in the database.

If we don't want that, the migration is useless and I can remove it. As you prefer.

**Comment by humitos** [comment]
> In MkDocs >= 1.0, this will never be a problem as theme is required. Therefore the default will never be used.

Why the default will never be used? Because it's mandatory to be a dict like `theme.name`? If they were using `mkdocs>=1.0` they need to be setting the theme manually otherwise, it shouldn't be working right now.

**Comment by davidfischer** [comment]
> Why the default will never be used? Because it's mandatory to be a dict like theme.name? If they were using mkdocs>=1.0 they need to be setting the theme manually otherwise, it shouldn't be working right now.

This is all correct. In 1.0+, theme.name is mandatory. Nobody is using 1.0+ unless they specify it in their requirements file.

File: readthedocs/projects/migrations/0028_feature-flag-mkdocs-theme.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocs/projects/migrations/0028_feature-flag-mkdocs-theme.py]: Yeah, this is not required for feature flags. It's only needed if we are setting a time-based default.

**Comment by agjohnson** [comment]
Great, sounds like this can be merged and hotfixed on our release branch.

**Pull request merged**